code/modules/html_interface/map/station_map.dm

```diff
@@ -0,0 +1,82 @@
+
+/proc/generateHoloMinimaps()
+	var/list/filters = list(
+		"deathsquad",
+		"ert",
+		"nukeops",
+		"elitesyndicate",
+		"vox",
+		)
+
+	for (var/f in filters)
+		centcommminimaps |= f
+		generateCentcommMinimap(f)
+
+	for (var/z = 1 to world.maxz)
+		holominimaps |= z
+		generateHoloMinimap(z)
+
+/proc/generateHoloMinimap(var/zLevel=1)
+	var/icon/canvas = icon('icons/480x480.dmi', "blank")
+
+	if(zLevel != map.zCentcomm)//The Centcomm Zlevel's minimap will remain blank for immersion purposes. Until we add a way to only draw certain areas depending on the suit's faction
+		for(var/i = 1 to 480)
```

Comment:
Byond doesn't actually do that, what it does is it expands the screen if you place an object on a screen loc further than world.view. So for example CENTER+world.view+1 is going to expand the screen in whatever direction 1 extra size.